server/src/test/java/org/elasticsearch/search/aggregations/bucket/prefix/IpPrefixAggregatorTests.java

```diff
@@ -0,0 +1,1075 @@
+            new TestIpDataHolder("2001:db8:a4ff:112a::7002:7ff2", "2001:db8:a4ff:112a::", prefixLength, 200),
+            new TestIpDataHolder("3007:db81:4b11:234f:1212:0:1:3", "3007:db81:4b11:234f::", prefixLength, 170),
+            new TestIpDataHolder("3007:db81:4b11:234f:7770:12f6:0:30", "3007:db81:4b11:234f::", prefixLength, 130)
+        );
+
+        // WHEN
+        testAggregation(aggregationBuilder, new MatchAllDocsQuery(), iw -> {
+            for (TestIpDataHolder ipDataHolder : ipAddresses) {
+                iw.addDocument(
+                    List.of(
+                        new SortedDocValuesField(ipField, new BytesRef(InetAddressPoint.encode(ipDataHolder.getIpAddress()))),
+                        new NumericDocValuesField(timeField, ipDataHolder.getTime())
+                    )
+                );
+            }
+        }, ipPrefix -> {
+            final Set<String> expectedSubnets = ipAddresses.stream()
+                .map(TestIpDataHolder::getSubnetAsString)
+                .collect(Collectors.toUnmodifiableSet());
+            final Set<String> ipAddressesAsString = ipPrefix.getBuckets()
+                .stream()
+                .map(InternalIpPrefix.Bucket::getKeyAsString)
+                .collect(Collectors.toUnmodifiableSet());
+
+            // THEN
+            ipPrefix.getBuckets().forEach(bucket -> {
+                assertTrue(bucket.isIpv6());
+                assertFalse(bucket.appendPrefixLength());
+                assertEquals(prefixLength, bucket.getPrefixLength());
+            });
+            assertFalse(ipPrefix.getBuckets().isEmpty());
+            assertEquals(expectedSubnets.size(), ipPrefix.getBuckets().size());
+            assertTrue(ipAddressesAsString.containsAll(expectedSubnets));
+            assertTrue(expectedSubnets.containsAll(ipAddressesAsString));
+
+            assertEquals(210, ((InternalSum) ipPrefix.getBuckets().get(0).getAggregations().get(subAggregationName)).getValue(), 0);
+            assertEquals(200, ((InternalSum) ipPrefix.getBuckets().get(1).getAggregations().get(subAggregationName)).getValue(), 0);
+            assertEquals(300, ((InternalSum) ipPrefix.getBuckets().get(2).getAggregations().get(subAggregationName)).getValue(), 0);
+        }, fieldTypes);
+    }
+
+    private Function<String, String> appendPrefixLength(int prefixLength) {
+        return subnetAddress -> subnetAddress + "/" + prefixLength;
+    }
+
+    private long defaultTime() {
+        return randomLongBetween(0, Long.MAX_VALUE);
+    }
+
+    private void testAggregation(
+        AggregationBuilder aggregationBuilder,
+        Query query,
+        CheckedConsumer<RandomIndexWriter, IOException> buildIndex,
+        Consumer<InternalIpPrefix> verify,
+        MappedFieldType... fieldTypes
+    ) throws IOException {
+        testCase(aggregationBuilder, query, buildIndex, verify, fieldTypes);
```

Comment:
Why are we wrapping `testCase` here?  Does this gain anything?

Reply:
This is probably just a result of a refactoring I did and didn't notice wrapping was useless.